src/parser/mod.rs

```diff
@@ -1258,23 +1258,28 @@ impl<'a> Parser<'a> {
                     }
                 };
                 self.expect_token(&Token::RParen)?;
-                if !self.consume_token(&Token::Period) {
-                    Ok(expr)
-                } else {
-                    let tok = self.next_token();
-                    let key = match tok.token {
-                        Token::Word(word) => word.to_ident(),
-                        _ => {
-                            return parser_err!(
-                                format!("Expected identifier, found: {tok}"),
-                                tok.location
-                            )
+                match self.try_parse_method(expr)? {
```

Comment:
left a suggestion to avoid the nested result, thinking with that it should be possible to return the expr as is and then here we avoid the need to match: e.g.
```rust
let expr = self.try_parse_method(expr)?;
if !self.consume_token(...) {
}
```